coramin/relaxations/multivariate.py

```diff
@@ -0,0 +1,117 @@
+from coramin.utils.coramin_enums import RelaxationSide, FunctionShape
+from coramin.relaxations.custom_block import declare_custom_block
+from coramin.relaxations.relaxations_base import BaseRelaxationData, ComponentWeakRef
+from pyomo.core.expr.visitor import identify_variables
+import math
+import pyomo.environ as pe
+from coramin.relaxations._utils import _get_bnds_list
+
+
+@declare_custom_block(name='MultivariateRelaxation')
+class MultivariateRelaxationData(BaseRelaxationData):
```

Comment:
Why is McCormick unrelated to multivariate?  I would have thought that McCormick would be a special case of multivariate.

Reply:
@jsiirola Good question. Multivariate is designed for multivariate convex or concave functions. If it is convex, we only do underestimators. If it is concave, we only do overestimators. Thus, `MultivariateRelaxation` does not do piecewise refinement. If the overestimators are needed for convex functions, then factorable programming has to be used to break the function into univariate and bivariate terms. This is distinctly different from the `McCormickRelaxation` which is designed for piecewise refinement.